### PR TITLE
Debug: sprinkle debug messages in thumbnailing code

### PR DESCRIPTION
**Pull request by bstoeger**
To ease trouble-shooting of the picture thumbnailer add a number
of debug- and info-messages.

Signed-off-by: Berthold Stoeger <bstoeger@mail.tuwien.ac.at>

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation
- [x] Debugging

### Pull request long description:

Add debug-messages to the thumbnailing code to help troubleshooting #1179

### CODE REVIEW

File: core/imagedownloader.cpp [hunk 1]
```diff
@@ -103,12 +105,18 @@ static void loadPicture(struct picture *picture, bool fromHash)
 SHashedImage::SHashedImage(struct picture *picture) : QImage()
 {
 	QUrl url = QUrl::fromUserInput(localFilePath(QString(picture->filename)));
-	if(url.isLocalFile())
+	if(url.isLocalFile()) {
```
Comment (neolit123) [inline core/imagedownloader.cpp]: might be a good opportunity to fix the missing space after `if`.

File: core/imagedownloader.cpp [hunk 2]
```diff
@@ -103,12 +105,18 @@ static void loadPicture(struct picture *picture, bool fromHash)
...
+		if (isNull())
+			qInfo() << "Failed loading picture " << url.toLocalFile();
+		else
+			qDebug() << "Loaded picture " << url.toLocalFile();
+	}
 	if (isNull()) {
 		// This did not load anything. Let's try to get the image from other sources
 		// Let's try to load it locally via its hash
 		QString filename = localFilePath(picture->filename);
+		qDebug() << "Translated filename: '" << picture->filename << "' --> '" << filename << "'";
```
Comment (neolit123) [inline core/imagedownloader.cpp]: probably better to use `arg()` here:
```
qDebug() << QStringLiteral("Translated filename: '%1' --> '%2'").arg(picture->filename).arg(filename);
```

**Review by neolit123 — CHANGES_REQUESTED** [review]
added a couple of small comments.

File: core/imagedownloader.cpp [hunk 3]
```diff
@@ -103,12 +105,18 @@ static void loadPicture(struct picture *picture, bool fromHash)
...
+		if (isNull())
+			qInfo() << "Failed loading picture " << url.toLocalFile();
+		else
+			qDebug() << "Loaded picture " << url.toLocalFile();
+	}
 	if (isNull()) {
 		// This did not load anything. Let's try to get the image from other sources
 		// Let's try to load it locally via its hash
 		QString filename = localFilePath(picture->filename);
+		qDebug() << QStringLiteral("Translated filename: %1 -> %2'").arg(picture->filename, filename);
```
Comment (neolit123) [inline core/imagedownloader.cpp]: there is a stray `'`.

File: core/imagedownloader.cpp [hunk 4]
```diff
@@ -119,9 +127,11 @@ SHashedImage::SHashedImage(struct picture *picture) : QImage()
 			load(filename);
 			if (!isNull()) {
 				// Make sure the hash still matches the image file
+				qDebug() << "Loaded picture from translated filename " << filename;
```
Comment (neolit123) [inline core/imagedownloader.cpp]: i think the space after `filename` is not needed as each `<<` append is separated by space?

Comment (neolit123) [reply core/imagedownloader.cpp]: applies to similar `qDebug()` calls?

File: core/imagedownloader.cpp [hunk 5]
```diff
@@ -44,6 +44,7 @@ bool ImageDownloader::loadFromUrl(const QUrl &url)
 		connect(&manager, &QNetworkAccessManager::finished, this,
 			[this,&success] (QNetworkReply *reply) { saveImage(reply, success); });
 		connect(&manager, &QNetworkAccessManager::finished, &loop, &QEventLoop::quit);
+		qDebug() << "Downloading image from " << url;
```
Comment (neolit123) [inline core/imagedownloader.cpp]: [1] sorry, my comment was about the extra space being redundant for all of these `qDebug()` calls.
```qDebug() << "Downloading image from" << url;```

File: core/imagedownloader.cpp [hunk 6]
```diff
@@ -68,6 +69,7 @@ void ImageDownloader::saveImage(QNetworkReply *reply, bool &success)
 		dir.mkpath(path);
 	QFile imageFile(path.append("/").append(hash.result().toHex()));
 	if (imageFile.open(QIODevice::WriteOnly)) {
+		qDebug() << "Write image to " << imageFile.fileName();
```
Comment (neolit123) [inline core/imagedownloader.cpp]: [1]

File: qt-models/divepicturemodel.cpp [hunk 7]
```diff
@@ -23,6 +23,7 @@ static void scaleImages(PictureEntry &entry, int size, int maxSize)
 	// If thumbnails were written by an earlier version, they might be smaller than needed.
 	// Rescale in such a case to avoid resizing artifacts.
 	if (thumbnail.isNull() || (thumbnail.size().width() < maxSize && thumbnail.size().height() < maxSize)) {
+		qDebug() << "No thumbnail in cache for " << entry.filename;
```
Comment (neolit123) [inline qt-models/divepicturemodel.cpp]: [1]

Comment (neolit123) [reply qt-models/divepicturemodel.cpp]:
```
qDebug() << "No thumbnail in cache for" << entry.filename;
```

**Review by neolit123 — APPROVED** [review]

**Comment by neolit123** [comment]
pushed a rebase to fix the remaining formatting issue on `qt-models/divepicturemodel.cpp:26`

**Comment by sfuchs79** [comment]
@neolit123 
I'm not 100% sure if it was @bstoeger intention to have this code in master ;-)
What I want to say: This code ALWAYS does produce a ton of debug messages as soon as you do something around dive pictures. Finally we may want to have s.th. like `#if DEBUG` around the debug code.
BTW: Maybe you could please once run the Windows build and do a test with adding a larger number of dive pictures to a dive? Sooner or later it would be very important to see if s.o. can reproduce my results from #1179

**Comment by neolit123** [comment]
@bstoeger @sfuchs79 
i trust you to fix this it if you think that there is way too much debug output.
we also support a flag for verbosity level.

**Comment by bstoeger** [comment]
> The image data was invalid, and QImageReader was unable to read an image from it. The can happen if the image file is damaged.

I'm not sure myself. ;) It shouldn't produce too much output in release mode (`qDebug()` doesn't produce output for me). We can remove the redundant stuff once we fixed the problems.

**Pull request merged**